hazelcast/src/test/java/com/hazelcast/map/IssuesTest.java

```diff
@@ -310,6 +314,74 @@ public void testMapClearDoesNotTriggerEqualsOrHashCodeOnKeyObject() {
         assertFalse("equals method should not have been called on key during clear", CompositeKey.equalsCalled);
     }
 
+    @Test
+    public void testIssue18079IndexDeserializationNoDuplicate() {
```

Comment:
Feels like `IssuesTest` is too generic class for this test (goes back to 2013 when I guess we put all sorts of tests related to different modules in a single class). Could you put somewhere in `com.hazelcast.map.impl.query`, maybe `QueryIndexTest` (maybe there's an even better one, not sure).